google/cloud/storage/internal/grpc/configure_client_context.cc

```diff
@@ -36,8 +37,10 @@ void AddIdempotencyToken(grpc::ClientContext& ctx,
 void ApplyRoutingHeaders(
     grpc::ClientContext& context,
     storage::internal::InsertObjectMediaRequest const& request) {
-  context.AddMetadata("x-goog-request-params",
-                      "bucket=projects/_/buckets/" + request.bucket_name());
+  context.AddMetadata(
+      "x-goog-request-params",
+      "bucket=projects%2F_%2Fbuckets%2F" +
+          google::cloud::internal::UrlEncode(request.bucket_name()));
```

Comment:
Yuck.  Can we just do `UrlEncode("projects/_/buckets/" + request.bucket_name())` ?

Reply:
Done.